backend/instance.ts

```diff
@@ -86,12 +87,12 @@ export class Instances {
     });
   }
 
-  add(): Instance {
-    this.currentPort++;
-    const port = this.currentPort;
+  async add(): Promise<Instance> {
+    const port = await detectPort(this.currentPort + 1);
```

Comment:
This works quite nicely actually, so if you start e.g. `webxdc-dev run /path/to/some.xdc` in one console (would pick port 7000 and also have port 7001 and 7002 for the instances) and then do `webxdc-dev run --port 7002 /path/to/some.xdc` then the backend would start at 7003 with two instances at 7004 and 7005.

If you then click `Add Instance` in the first window, then the next instance will get port 7006:

![image](https://github.com/user-attachments/assets/9ba623e9-f74d-40af-b90d-54aa96cb19a4)